### DISCUSSION TITLE
Dual license as Apache v2 and MIT

### DISCUSSION THREAD

**Discussion by gaul** [aws/aws-sdk-cpp#3941] [General]
### Describe the feature

Distribute the C++ SDK under two liberal licenses.

### Use Case

s3fs is licensed as a GPLv2 or later program.  GPLv2 is not compatible with Apache v2 and thus the license effectively upgrades to GPLv3 or later.

### Proposed Solution

Dual license the SDK like Rust does:

https://www.rust-lang.org/policies/licenses

### Other Information

Some discussion about Apache and GPL licenses:

https://www.apache.org/licenses/GPL-compatibility.html

Further StackOverflow discussion which states and incompatibility with GPLv2:

https://opensource.stackexchange.com/questions/1357/can-i-link-a-apache-2-0-library-into-software-under-gplv2

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

**Comment by jmklix** [comment]
Thanks for this suggestion, we are looking into this. I will make sure to include any updates here.

**Comment by rvansa** [comment]
Hello, has there been any conclusion from the considerations?

**Comment by kai-ion** [comment]
Converting this to a Discussion. It's an improvements we plan to pursue over time rather than one actionable change with a near-term milestone, so a Discussion better fits collecting the specifics and design input.
